fix GetVolumeInUse() function

NickrenREN opened the pull request:
Since we just want to get volume name info, each volume name just need to added once.  desiredStateOfWorld.GetVolumesToMount() will return volume and pod binding info,
if one volume is mounted to several pods, the volume name will be return several times. That is not what we want in this function.
We can add a new function to only get the volume name info  or judge whether the volume name is added to the desiredVolumesMap array.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/38443)
<!-- Reviewable:end -->

NickrenREN commented:
@dims  @saad-ali  Could you please ask k8s-bot to test ?

resouer commented:
@k8s-bot ok to test

NickrenREN commented:
@resouer  two checks failed. Maybe not related to the codes i modified. Could you please ask bot to test the two again?

Commit:
fix GetVolumeInUse() function
Since we just want to get volume name info, each volume name just need to added once. desiredStateOfWorld.GetVolumesToMount() will return volume and pod binding info,
if one volume is mounted to several pods, the volume name will be return several times. That is not what we want in this function.
We can add a new function to only get the volume name info or judge whether the volume name is added to the desiredVolumesMap array.

NickrenREN commented:
@resouer  two checks failed . But not related to the code I modified. Could you please ask bot to test the two again? thanks

NickrenREN commented:
@vishh  @resouer  @saad-ali   checks passed . Please review

NickrenREN commented:
Checks passed.  @jsafrane  @vishh  @saad-ali  Could one of you help with the review?

saad-ali commented:
/lgtm

saad-ali commented:
/approve

saad-ali commented:
Please cherry-pick to 1.5 once this is merged.

saad-ali commented:
@k8s-bot test this

NickrenREN commented:
@k8s-bot verify test this

saad-ali commented:
@k8s-bot cvm gke e2e test this

saad-ali commented:
@k8s-bot node e2e test this
